dhcpv4/ztpv4/parse_circuitid.go

```diff
@@ -0,0 +1,98 @@
+package ztpv4
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+)
+
+// CircuitID represents the structure of network vendor interface formats
+type CircuitID struct {
+	Slot    string
+	Module  string
+	Port    string
+	SubPort string
+	Vlan    string
+}
+
+var circuitRegexs = []*regexp.Regexp{
+	// Juniper QFX et-0/0/0:0.0
+	regexp.MustCompile("^et-(?P<slot>[0-9]+)/(?P<mod>[0-9]+)/(?P<port>[0-9]+):(?P<subport>[0-9]+).*$"),
+	// Juniper PTX et-0/0/0.0
+	regexp.MustCompile("^et-(?P<slot>[0-9]+)/(?P<mod>[0-9]+)/(?P<port>[0-9]+).(?P<subport>[0-9]+)$"),
+	// Arista Ethernet3/17/1
+	regexp.MustCompile("^Ethernet(?P<slot>[0-9]+)/(?P<mod>[0-9]+)/(?P<port>[0-9]+)$"),
+	// Juniper QFX et-1/0/61
+	regexp.MustCompile("^et-(?P<slot>[0-9]+)/(?P<mod>[0-9]+)/(?P<port>[0-9]+)$"),
+	// Arista Ethernet14:Vlan2001
+	// Arista Ethernet10:2020
+	regexp.MustCompile("^Ethernet(?P<port>[0-9]+):(?P<vlan>.*)$"),
+	// Cisco Gi1/10:2020
+	regexp.MustCompile("^Gi(?P<slot>[0-9]+)/(?P<port>[0-9]+):(?P<vlan>.*)$"),
+	// Nexus Ethernet1/3
+	regexp.MustCompile("^Ethernet(?P<slot>[0-9]+)/(?P<port>[0-9]+)$"),
+	// Juniper bundle interface ae52.0
+	regexp.MustCompile("^ae(?P<port>[0-9]+).(?P<subport>[0-9])$"),
+}
+
+// ParseCircuitID will parse dhcpv4 packet and return CircuitID info
+func ParseCircuitID(packet *dhcpv4.DHCPv4) (*CircuitID, error) {
+	// CircuitId info is stored as sub-option field in RelayAgentInformationOption
+	relayOptions := packet.RelayAgentInfo()
+
+	if relayOptions == nil {
+		return nil, fmt.Errorf("No relay agent information option found in the dhcpv4 pkt")
+	}
+
+	if relayOptions.Options == nil {
```

Comment:
I think this is never going to be nil, instead I would do:

```
optCircuit := relayOptions.Options.Get(dhcpv4.AgentCircuitIDSubOption)

if len(optCircuit) == 0 {
    return nil, fmt.Errorf("No relay agent information suboptions found in the dhcpv4 pkt")
}
```

Reply:
I think I am already checking this in [here](https://github.com/akshaynawale/dhcp/blob/master/dhcpv4/ztpv4/parse_circuitid.go#L55). As the Options is a Map[uint][]byte it will return no bytes for option code 1 if circuitid is not present. And string converts it to "".